npm/pkg/dataplane/policies/chain-management_linux.go

```diff
@@ -294,6 +243,235 @@ func (pMgr *PolicyManager) bootup(_ []string) error {
 	return nil
 }
 
+// detectIptablesVersion sets the global iptables variable to nft if detected or legacy if detected.
+// NPM will crash if it fails to detect either.
+// This global variable is referenced in all iptables related functions.
+func (pMgr *PolicyManager) detectIptablesVersion() error {
+	klog.Info("first attempt detecting iptables version. running: iptables-nft-save -t mangle")
+	cmd := pMgr.ioShim.Exec.Command(util.IptablesSaveNft, "-t", "mangle")
+	output, err := cmd.CombinedOutput()
+	if err == nil && strings.Contains(string(output), "KUBE-IPTABLES-HINT") || strings.Contains(string(output), "KUBE-KUBELET-CANARY") {
```

Comment:
Just in case, you might want to have braces for both strings contain check. SO it wont take the OR alone and forget to check err, unless you wanted it that way ?

Reply:
This is why I made the point of separating out the err != nil check first. It clutters the logic here and doesn't adhere to the [line of sight](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88) principle.

Reply:
Good call outs. if-else to the rescue